text/078-tbc-title.md

```diff
@@ -0,0 +1,508 @@
+
+### Secondary Goals
+
+- **External reuse** - Is there a pathway toward the components that are built to be used in isolation of styles and in isolation of even Wagtail itself (e.g. as a importable module).
+- **ES6 Modules Compatible** - There should be an easy migration path to adoption of ES6 modules.
+
+### History & Initial Research
+
+Throughout 2022, Wagtail has been releasing a major UI overhaul of the user interface and features in Wagtail’s page editor.
+
+As new UI components have been built, there is an understanding that jQuery is not an option, but React is not suitable for content that needs to be driven mostly by content in Django templates. This has lead to extremely ad-hoc approaches for UI elements, there is almost no consistency between how the new breadcrumbs, tabs and tooltips have been built. This makes it extremely hard for new developers to get started and core developers to maintain frontend code.
+
+React is an incredible library but better serves situations where the full control of the DOM branch is given to React, DOM rendering is 'given over' to JavaScript and Django templates cannot be leveraged (easily) to change what is inside those DOM elements.
+
+jQuery has served a great purpose, providing easy DOM traversal and manipulation, however it is becoming a bottleneck for new features and modern DOM APIs are sufficient and jQuery does not solve the problem of making it easy to write consistent code (component state). Additional to this we end up having to a lot of 'initialising' of elements whenever added to the DOM by other means (other JS or maybe async HTML).
+
+This lead us (UI team) to do initial research at the start of 2022 into what frameworks exist (based on suggestions from the Wagtail community). See [Lightweight JavaScript Framework research](https://github.com/lb-/bakerydemo/blob/ui-experiments/lightweight-frontend-framework-investigation-2022.md).
+
+This also lead to various rounds of feedback about what approaches could work with a leaning towards Stimulus. See the following links:
+
+- [main discussion sub-tread on a lightweight framework](https://github.com/wagtail/wagtail/discussions/7689#discussioncomment-2037913)
+- [page editor initial discussion on a lightweight framework](https://github.com/wagtail/wagtail/discussions/7739#discussioncomment-2022909)
+- [page editor discussion resulting in suggestions for frameworks](https://github.com/wagtail/wagtail/discussions/7739#discussioncomment-1926410)
+- [feedback from Matt](https://github.com/wagtail/wagtail/discussions/7689#discussioncomment-2775453)
+
+## Specification
+
+### Initial Implementation Overview
+
+- Install Stimulus `npm install @hotwired/stimulus --save`
+- Due to Stimulus being an ES6 module and Wagtail's codebase transpiling to ES5, TypeScript/Webpack and Jest will need to be configured to ensure that this library within node modules is to be transpiled also.
```

Comment:
Now we no longer support IE11, do we need the ES5 transpilation? If not, maybe we could consider removing the ES5 transpilation as part of this?

Reply:
I have given it a go a few times the last few months and keep running into large hurdles with Typescript. 

Might need some others in the core team to give it a go. 

My thinking is small steps toward the goal is good and including ES6 compilation in this RFC may be a bit too much.